File: app/src/main/java/com/peakmain/basicui/view/RadioCancelButton.java
package com.peakmain.basicui.view;

import android.content.Context;
import android.util.AttributeSet;

/**
 * author ：Peakmain
 * createTime：2020/3/10
 * mail:2726449200@qq.com
 * describe：
 */
public class RadioCancelButton extends android.support.v7.widget.AppCompatRadioButton {
    public RadioCancelButton(Context context) {
        super(context);
    }

    public RadioCancelButton(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public RadioCancelButton(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    @Override
    public void toggle() {
        setChecked(!isChecked());
    }
}

File: app/src/main/java/com/peakmain/basicui/view/RadioCancelButton.kt
package com.peakmain.basicui.view

import android.content.Context
import android.util.AttributeSet
import androidx.appcompat.R
import androidx.appcompat.widget.AppCompatRadioButton

/**
 * author ：Peakmain
 * createTime：2020/3/10
 * mail:2726449200@qq.com
 * describe：
 */
class RadioCancelButton @JvmOverloads constructor(context: Context?, attrs: AttributeSet?=null, defStyleAttr: Int= R.attr.radioButtonStyle) :
    AppCompatRadioButton(context,attrs, defStyleAttr){
    override fun toggle() {
        isChecked = !isChecked
    }
}